variables.tf

```diff
@@ -63,19 +63,17 @@ variable "location" {
   nullable    = false
 }
 
-variable "management_adapters" {
-  type        = list(string)
-  description = "A list of management adapters."
-  nullable    = false
-}
-
 variable "name" {
   type        = string
   description = "The name of the HCI cluster. Must be the same as the name when preparing AD."
 
   validation {
-    condition     = length(var.name) < 16 && length(var.name) > 0
-    error_message = "value of name should be less than 16 characters and greater than 0 characters"
+    condition     = var.cluster_name != "" || (length(var.name) <= 16 && length(var.name) > 0)
```

Comment:
<16

Reply:
seems not.
When cluster_name is ""，we will use var.name as var.cluster_name, so we need to ensure its length < 16